src/cargo/ops/registry.rs

```diff
@@ -55,6 +57,39 @@ pub fn publish(manifest_path: &Path,
     Ok(())
 }
 
+fn check_git_cleanliness(current_path: &Path) -> CargoResult<()>{
+    match git2::Repository::open(current_path){
```

Comment:
I forget the semantics of `open`, unfortunately, but does this discover the git repo hierarchically as well? It may be the case that we're publishing a sub-crate of a git project, so we may want to traverse upwards a bit looking for the rot.